fuse_models/include/fuse_models/unicycle_2d_state_cost_functor.h

```diff
@@ -163,19 +163,21 @@ bool Unicycle2DStateCostFunctor::operator()(
     acc_linear1,
     T(dt_),
     delta_position_pred,
-    yaw_pred,
+    delta_yaw_pred,
     vel_linear_pred,
     vel_yaw_pred,
     acc_linear_pred);
 
   // rotate the position residual into the local frame
-  T sin_pred_inv = ceres::sin(-yaw_pred[0]);
-  T cos_pred_inv = ceres::cos(-yaw_pred[0]);
+  const Eigen::Matrix<T, 2, 1> position1_map(position1[0], position1[1]);
+  const Eigen::Matrix<T, 2, 1> position2_map(position2[0], position2[1]);
+  const Eigen::Matrix<T, 2, 1> delta_position = fuse_core::rotationMatrix2D(-yaw1[0]) * (position2_map - position1_map);
+  const T delta_yaw = yaw2[0] - yaw1[0];  // omitting fuse_core::wrapAngle2D because it is done later on
+  const Eigen::Matrix<T, 2, 1> delta_position_pred_map(delta_position_pred[0], delta_position_pred[1]);
```

Comment:
These aren't really Eigen maps, but I just realized you can use `position1` because it's already taken.

Just highlighting that. I don't really have a better option, so it's ok for me. Maybe others have a better suggestion :thinking: